### DISCUSSION TITLE
Confidence interval for Harrell's c-index?

### DISCUSSION THREAD

**Discussion by KenleyData** [kogalur/randomForestSRC#134] [Q&A]
I am using the RandomForestSRC package to develop a Random Survival Forest model. The metric we are most interested in is the Harrell's c-index when the trained forest is applied to a testing dataset. 

So far I have been using the rcorr.cens function to create a c-index from the RandomForestSRC predictions (code pasted below). However, I would like to have a confidence interval for the c-index. Since randomForestSRC performs confidence intervals for variable importance, I am wondering if there is a way to do 95% confidence intervals for this statistic: the Harrell's c-index for trained forest applied to test set.

Thanks in advance for any advice.

EXISTING CODE:
rsf_t <- rfsrc(formula_rf_t, data = early_cohort, ntree=1000, mtry=20, nodesize=5, seed=1)
survival_predictions_t <- predict(rsf_t, newdata = late_cohort)
rcorr.cens(-survival_predictions_t$predicted , Surv(t_outcome_late,ev_outcome_late))["C Index"]

**Comment by ishwaran** [comment]
One way to proceed would be to use .632 bootstrap.  Put a loop around the entire code with each call to `rfsrc` using a .632 bootstrap of the data.  So each iteration of the loop would have a call something like:

```
n <- nrow(early_cohort)
rsf_t <- rfsrc(formula_rf_t, 
                       data=early_cohort[sample(1:n, size=.632*n, replace = FALSE), drop = FALSE],
                       ntree=1000, mtry=20, nodesize=5, seed=1)
survival_predictions_t <- predict(rsf_t, newdata = late_cohort)
rcorr.cens(-survival_predictions_t$predicted , Surv(t_outcome_late,ev_outcome_late))["C Index"]

```

A second way to proceed would be to use a subsampling approach as used by the function `subsample`.   This method is used for estimating standard error and confidence intervals for variable importance (VIMP) but it is easily modified to do the same for the internal error rate (the so-called out-of-bag, OOB, error rate).   Then using jackknife for example, one could calculate the standard error and confidence interval for the error rate.  Note this does not even need to use the test data.
